### PR TITLE
creOnic added

### PR DESCRIPTION
**Pull request by noclew**
## Description

Added a keyboard, creOnic, which is a hobby keyboard that I made. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ x ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ x  ] I have updated the documentation accordingly.
- [ x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ x ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/nckiibs/creonic/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    _LOWER,
+    _RAISE,
+    _FN
+};
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    LOWER = SAFE_RANGE,
+    RAISE
+};
```
Comment (zvecr) [inline keyboards/nckiibs/creonic/keymaps/default/keymap.c:30]:
```suggestion
#define LOWER   MO(_LOWER)
#define RAISE   MO(_RAISE)
```

File: keyboards/nckiibs/creonic/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+ * | Tab  |   Q  |   W  |   E  |   R  |   T  |   Y  |   U  |   I  |   O  |   P  | Del  |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * | Esc  |   A  |   S  |   D  |   F  |   G  |   H  |   J  |   K  |   L  |   ;  |Enter |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * | Shift|   Z  |   X  |   C  |   V  |   B  |   N  |   M  |   ,  |   .  |   /  |  "   |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | Brite| Ctrl | Alt  | GUI  |Lower | Space|Space |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ * */
+    [_BASE] = LAYOUT(
```
Comment (zvecr) [inline keyboards/nckiibs/creonic/keymaps/via/keymap.c:46]:
```suggestion
    [_BASE] = LAYOUT_ortho_5x12(
```

File: keyboards/nckiibs/creonic/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+ * |   ~  |   !  |   @  |   #  |   $  |   %  |   ^  |   &  |   *  |   (  |   )  | Del  |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * | Del  |  F1  |  F2  |  F3  |  F4  |  F5  |  F6  |   _  |   +  |   {  |   }  |  |   |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * |      |  F7  |  F8  |  F9  |  F10 |  F11 |  F12 |ISO ~ |ISO | |      |      |      |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_LOWER] = LAYOUT( \
```
Comment (zvecr) [inline keyboards/nckiibs/creonic/keymaps/via/keymap.c:67]:
```suggestion
[_LOWER] = LAYOUT_ortho_5x12( \
```

File: keyboards/nckiibs/creonic/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+ * |   `  |   1  |   2  |   3  |   4  |   5  |   6  |   7  |   8  |   9  |   0  | Del  |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * | Del  |  F1  |  F2  |  F3  |  F4  |  F5  |  F6  |   -  |   =  |   [  |   ]  |  \   |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * |      |  F7  |  F8  |  F9  |  F10 |  F11 |  F12 |ISO # |ISO / |  PGUP|  PGDN|      |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_RAISE] = LAYOUT( \
```
Comment (zvecr) [inline keyboards/nckiibs/creonic/keymaps/via/keymap.c:88]:
```suggestion
[_RAISE] = LAYOUT_ortho_5x12( \
```

File: keyboards/nckiibs/creonic/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+ * |      | Reset|      |      |      |      |      |      |      |      |      |  Del |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * |      |      |      |Aud on|AudOff|AGnorm|AGswap|Qwerty|Colemk|Dvorak|      |      |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * |      |Voice-|Voice+|Mus on|MusOff|MidiOn|MidOff|      |      |      |      |      |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |             |      |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_FN] = LAYOUT( \
```
Comment (zvecr) [inline keyboards/nckiibs/creonic/keymaps/via/keymap.c:109]:
```suggestion
[_FN] = LAYOUT_ortho_5x12( \
```

File: keyboards/nckiibs/creonic/config.h [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+  #define RGBLIGHT_EFFECT_CHRISTMAS
+  #define RGBLIGHT_EFFECT_STATIC_GRADIENT
+  #define RGBLIGHT_EFFECT_RGB_TEST
+  #define RGBLIGHT_EFFECT_ALTERNATING
+/*== customize breathing effect ==*/
+  /*==== (DEFAULT) use fixed table instead of exp() and sin() ====*/
+  #define RGBLIGHT_BREATHE_TABLE_SIZE 256      // 256(default) or 128 or 64
+  /*==== use exp() and sin() ====*/
+  #define RGBLIGHT_EFFECT_BREATHE_CENTER 1.85  // 1 to 2.7
+  #define RGBLIGHT_EFFECT_BREATHE_MAX    255   // 0 to 255
```
Comment (zvecr) [inline keyboards/nckiibs/creonic/config.h:85]:
```suggestion
```

File: keyboards/nckiibs/creonic/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+    _LOWER,
+    _RAISE,
+    _FN
+};
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    LOWER = SAFE_RANGE,
+    RAISE
+};
```
[the remark this answers is not in the document]
Comment (noclew) [reply keyboards/nckiibs/creonic/keymaps/via/keymap.c:30]: Hello, thanks for the help. Where can I a reference to `FN_MO` keycode? I'd like to understand how it works, but could not find a reference from the documentation.

File: keyboards/nckiibs/creonic/keymaps/default/keymap.c (hunk 1 above)
Comment (noclew) [reply keyboards/nckiibs/creonic/keymaps/default/keymap.c:30]: What is the difference between MO() and FN_MO that you used above?

File: keyboards/nckiibs/creonic/keymaps/via/rules.mk [hunk 8]
```diff
@@ -0,0 +1,2 @@
+VIA_ENABLE = yes
+BOOTMAGIC_ENABLE = lite
```
[the remark this answers is not in the document]
Comment (noclew) [reply keyboards/nckiibs/creonic/keymaps/via/rules.mk:2]: I wanted to test out VIA, but VIA software could not detect my keyboard although I added VIA_ENABLE = yes. Can you let me know if there are missing pieces that I need to add more to enable VIA in my codes?

File: keyboards/nckiibs/creonic/readme.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
+# creonic
+
+![creonic](imgur.com image replace me!)
+
+A short description of the keyboard/project
+
+* Keyboard Maintainer: [noclew](https://github.com/yourusername)
+* Hardware Supported: The PCBs, controllers supported
+* Hardware Availability: Links to where you can find this hardware
```
[the remark this answers is not in the document]
Comment (noclew) [reply keyboards/nckiibs/creonic/readme.md]: done!

**Comment by noclew** [comment]
Hi, thanks for your wonderful help for this noob! I made all the changes and pushed them. By the way can you please let me know where I can find a reference regarding FN_MO keycoade?

**Pull request closed without merging**